### DISCUSSION TITLE
Two columns with reference to same table

### DISCUSSION THREAD

**Discussion by mmsgov** [prisma/prisma#17004] [Q&A] [answered]

**Comment by pantharshit00** [comment]
I am glad that you were able to figure it out. 

You can post your answer here in separate thread and mark it as an answer. That way other people can also find the solution.

**Comment by mmsgov** [comment, marked as answer]
A reference to a discussion with relevant explanation - [issue 1483](https://github.com/prisma/prisma/issues/1483)

Prisma code with solution:
```prisma 
model User {
  id                  Int           @id @default(autoincrement())
  clientAppointment   Appointment[] @relation("clientUser")
  providerAppointment Appointment[] @relation("providerUser")
}

model Appointment {
  id             Int   @id @default(autoincrement())
  clientUser     User  @relation("clientUser", fields: [clientUserId], references: [id])
  clientUserId   Int
  providerUser   User? @relation("providerUser", fields: [providerUserId], references: [id])
  providerUserId Int
}
```
Thanks @pantharshit00

**Reply by bananabrann** [reply]
Huh, oddly enough this appears to me not be working for my situation

```prisma
// other fields removed
model Lot {
  id              Int         @id @default(autoincrement())
  editedBy        User?       @relation("editByUser", fields: [editedByUserId], references: [id])
  createdBy       User        @relation("createdByUser", fields: [createdByUserId], references: [id])
  editedByUserId  String
  createdByUserId String
}
model User {
  id            String    @id
  editedByUser  User[]    @relation("editedByUser")
  createdByUser User[]    @relation("createdByUser")
}
```


> Error validating field `editedBy` in model `Lot`: The relation field `editedBy` on model `Lot` is missing an opposite relation field on the model `User`. Either run `prisma format` or add it manually.

Running `prisma format` adds `User` to `Lot` fields, which only messes it up further. Same error 🤔 Any ideas?

**Reply by bananabrann** [reply]
D'oh. `editByUser` is a typo --it doesn't match the field `editedByUser` in `User` 😒

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
